File: 2460. Apply Operations to an Array/ApplyOperationsToAnArray.java
public class ApplyOperationsToAnArray {
    public int[] applyOperations(int[] nums) {
        int n = nums.length, i, count = 0;
        for (i=0;i<n-1;i++){
            if (nums[i] == 0){
                count++;
                continue;
            }
            if (nums[i] == nums[i+1]){
                nums[i] *= 2;
                nums[i+1] = 0;
            }
            nums[i-count] = nums[i];
            if(count!=0){
                nums[i] = 0;
            }
        }
        nums[i-count] = nums[n-1];
        if (count!=0){
            nums[n-1] = 0;
        }

        return nums;
    }
}

File: 2460. Apply Operations to an Array/Apply Operations to an Array.py
nums = [1,2,2,4]
n = len(nums)
count = 0
for i in range(n-1):
    if not nums[i]:
        count += 1
        continue
    if nums[i] == nums[i+1]:
        nums[i] *= 2
        nums[i+1] = 0
    nums[i-count],nums[i] = nums[i],nums[i-count]
nums[i+1-count],nums[-1] = nums[-1],nums[i+1-count]
print(nums)